modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1763,13 +1766,190 @@ def sort_by(
+                Where q is the proportion we sample in scenario 3, k is the proportion we sample
+                of remaining rows in scenario 2, and A is the threshold below which we just return
+                the entire dataframe, instead of sampling (scenario 1).
+                q = 0.1 and k = 0.05 were picked such that this formula holds for A = 100.
+                """
+                quantiles = [
+                    i / (NPartitions.get() * 2) for i in range(NPartitions.get() * 2)
+                ]
+                # Heuristic for a "small" df we will compute quantiles over entirety of.
+                if len(df) <= A:
+                    return df, np.quantile(df[columns[0]], quantiles, method=method)
+                # Heuristic for a "medium" df where we will include first 100 (A) rows, and sample
+                # of remaining rows when computing quantiles.
+                if len(df) <= A * (1 - k) / (1 - q):
+                    return df, np.quantile(
+                        np.concatenate(
+                            (
+                                df[columns[0]][:100].values,
+                                df[columns[0]][100:].sample(frac=k),
+                            )
+                        ),
+                        quantiles,
+                        method=method,
+                    )
+                # Heuristic for a "large" df where we will sample 10% (q) of all rows to compute quantiles
+                # over.
+                return df, np.quantile(
+                    df[columns[0]].sample(frac=q), quantiles, method=method
+                )
+
+            def pivot_func(samples):
+                """
+                Determine quantiles from the given samples.
+
+                This function takes as input the quantiles calculated over all partitions from
+                `sample_func` defined above, and determines a final NPartitions.get() * 2 quantiles
+                to use to roughly sort the entire dataframe. It does so by collating all the samples
+                and computing NPartitions.get() * 2 quantiles for the overall set.
+                """
+                # We don't call `np.unique` on the samples, since if a quantile shows up in multiple
+                # partition's samples, this is probably an indicator of skew in the dataset, and we
+                # want our final partitions to take this into account.
+                all_pivots = np.array(samples).flatten()
+                quantiles = [
+                    i / (NPartitions.get() * 2) for i in range(NPartitions.get() * 2)
+                ]
+                overall_quantiles = np.quantile(all_pivots, quantiles, method=method)
+                if self.dtypes[columns[0]] != object:
+                    overall_quantiles[0] = np.NINF
+                    overall_quantiles[-1] = np.inf
+                return overall_quantiles
+
+            def split_func(df, pivots):
```

Comment:
Again, can we move this function definition outside of this method? It looks like it just needs `na_position`.